### PR TITLE
GL Backend Changes/Fixes

### PR DESCRIPTION
**Pull request by DrChat**
* Shaders:
 * Fix incorrect compare against bool consts
 * Always output the info log if it isn't empty (for shader warnings)
* GL backend:
 * Setup to allow Xenia to detect a graphics TDR on contexts
 * TODO: Dump the command stream on TDR.
 * Minor: Blitter::Draw asserts if src_texture is 0.

### CODE REVIEW

File: src/xenia/gpu/gl4/gl4_shader.cc [hunk 1]
```diff
@@ -135,18 +135,21 @@ bool GL4Shader::CompileProgram() {
     return false;
   }
 
+  // Output info log.
+  // log_length includes the null character.
+  GLint log_length = 0;
+  glGetProgramiv(program_, GL_INFO_LOG_LENGTH, &log_length);
+  std::string info_log;
+  info_log.resize(log_length - 1);
```
Comment (sephiroth99) [inline src/xenia/gpu/gl4/gl4_shader.cc]: When log_length is 0, this fails with "string too long" error.

Comment (DrChat) [inline src/xenia/gpu/gl4/gl4_shader.cc]: You using AMD or Nvidia?

Comment (sephiroth99) [inline src/xenia/gpu/gl4/gl4_shader.cc]: AMD...

Comment (DrChat) [inline src/xenia/gpu/gl4/gl4_shader.cc]: Okay, Nvidia always returns 1 if the log is empty I guess.

**Pull request merged**